rio_cogeo/cogeo.py

```diff
@@ -154,62 +179,71 @@ def cog_translate(
                 meta.pop("compress", None)
                 meta.pop("photometric", None)
 
-                with MemoryFile() as memfile:
-                    with memfile.open(**meta) as mem:
-                        wind = list(mem.block_windows(1))
-
-                        if not quiet:
-                            click.echo("Reading input: {}".format(src_path), err=True)
-                        fout = os.devnull if quiet else sys.stderr
-                        with click.progressbar(
-                            wind, length=len(wind), file=fout, show_percent=True
-                        ) as windows:
-                            for ij, w in windows:
-                                matrix = vrt_dst.read(window=w, indexes=indexes)
-                                mem.write(matrix, window=w)
-
-                                if add_mask or mask:
-                                    mask_value = vrt_dst.dataset_mask(window=w)
-                                    mem.write_mask(mask_value, window=w)
-
-                        if not quiet:
-                            click.echo("Adding overviews...", err=True)
-
-                        if overview_level is None:
-                            overview_level = get_maximum_overview_level(
-                                vrt_dst,
-                                min(
-                                    int(dst_kwargs["blockxsize"]),
-                                    int(dst_kwargs["blockysize"]),
-                                ),
-                            )
+                if in_memory is None:
+                    in_memory = vrt_dst.width * vrt_dst.height < IN_MEMORY_THRESHOLD
 
-                        overviews = [2 ** j for j in range(1, overview_level + 1)]
-                        mem.build_overviews(
-                            overviews, ResamplingEnums[overview_resampling]
+                with ExitStack() as ctx:
```

Comment:
You could pull the `WarpedVRT` into the `ExitStack` and save one level of indentation